internal/services/resource/resource_group_resource.go

```diff
@@ -209,3 +213,50 @@ https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs#features
 `, name, strings.Join(formattedResourceUris, "\n"))
 	return fmt.Errorf(strings.ReplaceAll(message, "'", "`"))
 }
+
+func resourceExistById(ctx context.Context, providersClient *resources.ProvidersClient, resourceClient *resources.Client, resourceId string) (exist bool, err error) {
+	id, err := parse.GenericResourceID(resourceId)
+	if err != nil {
+		return true, fmt.Errorf("parsing generic resource id %s: %v", resourceId, err)
+	}
+
+	resourceApiVersion, err := getResourceTypeApiVersion(ctx, providersClient, id.ResourceProvider, id.ResourceType)
+	if err != nil {
+		return true, err
+	}
+
+	resource, err := resourceClient.GetByID(ctx, resourceId, resourceApiVersion)
+	if err != nil {
+		if utils.ResponseWasNotFound(resource.Response) {
+			return false, nil
+		}
+		return true, err
+	}
+
+	return resource.ID != nil, nil
+}
+
+func getResourceTypeApiVersion(ctx context.Context, client *resources.ProvidersClient, providerName string, targetResourceType string) (apiVersion string, err error) {
+	provider, err := client.Get(ctx, providerName, fmt.Sprintf("resourceTypes/%s", targetResourceType))
+	if err != nil {
+		return "", err
+	}
+
+	if provider.ResourceTypes == nil {
+		return "", fmt.Errorf("getting resource(%s/%s) default api version, the provider returned nil on resource type. ", providerName, targetResourceType)
+	}
+
+	for _, resourceType := range *provider.ResourceTypes {
+		if resourceType.ResourceType != nil && *resourceType.ResourceType == targetResourceType {
+			if resourceType.DefaultAPIVersion != nil {
+				return *resourceType.DefaultAPIVersion, nil
+			} else if len(*resourceType.APIVersions) > 0 {
+				return (*resourceType.APIVersions)[0], nil
+			}
+
+			return "", fmt.Errorf("getting resource(%s/%s) api version, there is no avaiable api version. ", providerName, targetResourceType)
+		}
+	}
+
+	return "", fmt.Errorf("getting resource(%s/%s) api version, there is no match resource type. ", providerName, targetResourceType)
```

Comment:
This API is unfortunately incomplete. Historically, ARM has not been strict with manifest configuration - it's possible to expose child resources which are not declared in the manifest, and as such, not exposed by the `/providers` API. To give a concrete example, when I run `az provider show --namespace Microsoft.Storage --query resourceTypes[].resourceType`, I don't see `storageAccounts/blobServices/containers` (which is a valid resourceType) listed.

So if there is no API retrieved, can we fallback to the original listing logic?